app/frontend/components/shared/elective-tag.tsx

```diff
@@ -2,19 +2,19 @@ import { Tag, TagProps } from "@chakra-ui/react"
 import React from "react"
 import { useTranslation } from "react-i18next"
 
-interface ElectiveTagProps {
+interface ElectiveTagProps extends TagProps {
   hasElective?: boolean
 }
 
-export function ElectiveTag(props: Partial<ElectiveTagProps & TagProps>) {
+export const ElectiveTag: React.FC<ElectiveTagProps> = ({ hasElective, ...rest }) => {
   const { t } = useTranslation()
 
-  const isElective = t("requirementsLibrary.elective")
-  const hasElective = t("requirementsLibrary.hasElective")
+  const isElectiveText = t("requirementsLibrary.elective")
+  const hasElectiveText = t("requirementsLibrary.hasElective")
 
   return (
-    <Tag bg={"theme.yellowLight"} color={"text.secondary"} fontWeight={700} fontSize={"xs"} {...props}>
```

Comment:
spreading the react props causes a console error about using react props as DOM element props. extract out the relevant react prop first to prevent such errors.